contracts/Content/UniqueContent.sol

```diff
@@ -0,0 +1,173 @@
+        uint256 _version = ++uniqueAssetInfo[_uniqueId].version;
+
+        emit UniqueUriUpdated(_uniqueId, _version);
+    }
+
+    /**
+    * @dev Returns the original asset's receiver address and royalty amount for a token sold at a certain sales price
+    * @param _uniqueId uint256 ID of token to query
+    * @param _salePrice price the asset is to be purchased for
+    */
+    function royaltyInfo(uint256 _uniqueId, uint256 _salePrice) external view override returns (address receiver, uint256 royaltyAmount){
+        require(_exists(_uniqueId), "Unique Id does not exist");
+        // grabs the original asset's content contract address and tokenId
+        address originalAddress = uniqueAssetInfo[_uniqueId].contentAddress;
+        uint256 tokenId = uniqueAssetInfo[_uniqueId].tokenId;
+        // calls royaltyInfo() in the original asset's contract
+        (receiver, royaltyAmount) = IERC2981Upgradeable(originalAddress).royaltyInfo(tokenId, _salePrice);
+    }
+
+    /**
+    * @dev Returns an array of receiver addresses and royalty amounts for a token sold at a certain sales price
+    * @param _uniqueId uint256 ID of token to query
+    * @param _salePrice price the asset is to be purchased for
+    */
+    function multipleRoyaltyInfo(uint256 _uniqueId, uint256 _salePrice) external view override returns (address[] memory receivers, uint256[] memory royaltyAmounts) {
+        require(_exists(_uniqueId), "Unique Id does not exist");
+        uint24[] memory rates;
+        // grabs the arrays of receivers and rates for the token
+        (receivers, rates) = _getMultipleRoyalties(_uniqueId);
+        // calculates royaltyAmount for each receiver and places them in an array
+        royaltyAmounts = new uint256[](rates.length);
+        for (uint256 i = 0; i < receivers.length; ++i) {
+            royaltyAmounts[i] = _salePrice * rates[i] / 1e6;
+        }
+    }
+
+    /**
+     * @dev Sets a specific token's royalties
+     * @param _uniqueId uint256 ID of the token to set royalties of
+     * @param _royaltyReceivers array of addresses to receive the royalties
+     * @param _royaltyRates array of royalty fee percentages
+     */
+    function setTokenRoyalties(uint256 _uniqueId, address[] memory _royaltyReceivers, uint24[] memory _royaltyRates) external override {
+        require(_exists(_uniqueId), "Unique Id does not exist");
+        require(uniqueAssetInfo[_uniqueId].creator == _msgSender(), "You are not the creator of this token");
+        _verifyRoyalties(_royaltyReceivers, _royaltyRates);
+        delete tokenRoyalties[_uniqueId];
+        _setTokenRoyalties(_uniqueId, _royaltyReceivers, _royaltyRates);
+    }
+
+    // Interface support
+    function supportsInterface(bytes4 interfaceId) public view virtual override(ERC721Upgradeable, IERC165Upgradeable) returns (bool) {
+        return super.supportsInterface(interfaceId);
+    }
+
+    function onERC1155Received(
```

Comment:
I don't think you need to manually override onERC1155Received() and onERC1155BatchReceived(). Just inherit ERC1155HolderUpgradeable and it should come with these automatically.